components/renku_data_services/k8s_watcher/core.py

```diff
@@ -145,10 +151,65 @@ async def stop(self, timeout: timedelta = timedelta(seconds=10)) -> None:
                     continue
 
 
-def k8s_object_handler(cache: K8sDbCache) -> EventHandler:
+async def collect_metrics(
+    previous_obj: K8sObject | None,
+    new_obj: APIObjectInCluster,
+    user_id: str,
+    metrics: MetricsService,
+    rp_repo: ResourcePoolRepository,
+) -> None:
+    """Track product metrics."""
+    user = APIUser(id=user_id)
```

Comment:
suggestion(non-blocking): I think that the event type from the wathc would be useful here. But if we do this we should implement it in a follow up PR. If it is even possible.

Reply:
I've added it to use DELETED for session stopped. Not sure it's useful for the other metrics